Flaky tests with database backup in test_premium

## Problem Definition

Flaky tests seen in this run: https://travis-ci.org/github/rotki/rotki/jobs/682265131


```
=================================== FAILURES ===================================
__________ test_try_premium_at_start_old_account_can_pull_data[True] ___________

rotkehlchen_instance = <rotkehlchen.rotkehlchen.Rotkehlchen object at 0x7fd326c97978>
username = 'testuser', db_password = '123'
rotki_premium_credentials = <rotkehlchen.premium.premium.PremiumCredentials object at 0x7fd326c97d68>

    @pytest.mark.parametrize('start_with_valid_premium', [True])
    def test_try_premium_at_start_old_account_can_pull_data(
            rotkehlchen_instance,
            username,
            db_password,
            rotki_premium_credentials,
    ):
        setup_starting_environment(
            rotkehlchen_instance=rotkehlchen_instance,
            username=username,
            db_password=db_password,
            premium_credentials=rotki_premium_credentials,
            first_time=False,
            same_hash_with_remote=False,
            newer_remote_db=True,
            db_can_sync_setting=True,
        )
>       assert_db_got_replaced(rotkehlchen_instance=rotkehlchen_instance, username=username)

rotkehlchen/tests/integration/test_premium.py:213: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

rotkehlchen_instance = <rotkehlchen.rotkehlchen.Rotkehlchen object at 0x7fd326c97978>
username = 'testuser'

    def assert_db_got_replaced(rotkehlchen_instance: Rotkehlchen, username: str):
        """For environment setup with setup_starting_environment make sure DB is replaced"""
        msg = 'Test default main currency should be different from the restored currency'
        assert DEFAULT_TESTS_MAIN_CURRENCY != A_GBP, msg
        # At this point pulling data from rotkehlchen server should have worked
        # and our database should have been replaced. The new data have different
        # main currency
        assert rotkehlchen_instance.data.db.get_main_currency() == A_GBP
        # Also check a copy of our old DB is kept around.
        directory = os.path.join(rotkehlchen_instance.data.data_directory, username)
        files = [os.path.join(directory, f) for f in os.listdir(directory)]
        msg = f'Expected 2 files in the directory but got {files}'
>       assert len(files) == 2, msg
E       AssertionError: Expected 2 files in the directory but got ['/home/travis/.cache/.rotkehlchen-test-dir/testuser/rotkehlchen_db_2020_05_02_12_33_03.backup', '/home/travis/.cache/.rotkehlchen-test-dir/testuser/rotkehlchen.db', '/home/travis/.cache/.rotkehlchen-test-dir/testuser/dbinfo.json']

rotkehlchen/tests/utils/premium.py:207: AssertionError
travis_fold:start:py-9078.stderr-setup.0
---------------------------- Captured stderr setup -----------------------------
Exception ignored in: <function DBHandler.__del__ at 0x7fd339b691e0>
Traceback (most recent call last):
  File "/home/travis/build/rotki/rotki/rotkehlchen/db/dbhandler.py", line 194, in __del__
    dbinfo = {'sqlcipher_version': self.sqlcipher_version, 'md5_hash': self.get_md5hash()}
  File "/home/travis/build/rotki/rotki/rotkehlchen/db/dbhandler.py", line 245, in get_md5hash
    with open(filename, 'rb') as f:
FileNotFoundError: [Errno 2] No such file or directory: '/home/travis/.cache/.rotkehlchen-test-dir/testuser/rotkehlchen.db'
travis_fold:end:py-9078.stderr-setup.0
travis_fold:start:py-9078.log-setup.0
------------------------------ Captured log setup ------------------------------
WARNING  rotkehlchen.chain.ethereum.manager:manager.py:132 Could not connect to an ethereum node. Will use etherscan only
travis_fold:end:py-9078.log-setup.0
________ test_try_premium_at_start_old_account_can_pull_old_data[True] _________

rotkehlchen_instance = <rotkehlchen.rotkehlchen.Rotkehlchen object at 0x7fd327091518>
username = 'testuser', db_password = '123'
rotki_premium_credentials = <rotkehlchen.premium.premium.PremiumCredentials object at 0x7fd326d542b0>

    @pytest.mark.parametrize('start_with_valid_premium', [True])
    def test_try_premium_at_start_old_account_can_pull_old_data(
            rotkehlchen_instance,
            username,
            db_password,
            rotki_premium_credentials,
    ):
        """
        Test that if the remote DB is of an old version its upgraded before replacing our current
    
        For an old account
        """
        setup_starting_environment(
            rotkehlchen_instance=rotkehlchen_instance,
            username=username,
            db_password=db_password,
            premium_credentials=rotki_premium_credentials,
            first_time=False,
            same_hash_with_remote=False,
            newer_remote_db=True,
            db_can_sync_setting=True,
            remote_data=REMOTE_DATA_OLDER_DB,
        )
>       assert_db_got_replaced(rotkehlchen_instance=rotkehlchen_instance, username=username)

rotkehlchen/tests/integration/test_premium.py:239: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

rotkehlchen_instance = <rotkehlchen.rotkehlchen.Rotkehlchen object at 0x7fd327091518>
username = 'testuser'

    def assert_db_got_replaced(rotkehlchen_instance: Rotkehlchen, username: str):
        """For environment setup with setup_starting_environment make sure DB is replaced"""
        msg = 'Test default main currency should be different from the restored currency'
        assert DEFAULT_TESTS_MAIN_CURRENCY != A_GBP, msg
        # At this point pulling data from rotkehlchen server should have worked
        # and our database should have been replaced. The new data have different
        # main currency
        assert rotkehlchen_instance.data.db.get_main_currency() == A_GBP
        # Also check a copy of our old DB is kept around.
        directory = os.path.join(rotkehlchen_instance.data.data_directory, username)
        files = [os.path.join(directory, f) for f in os.listdir(directory)]
        msg = f'Expected 2 files in the directory but got {files}'
>       assert len(files) == 2, msg
E       AssertionError: Expected 2 files in the directory but got ['/home/travis/.cache/.rotkehlchen-test-dir/testuser/rotkehlchen.db', '/home/travis/.cache/.rotkehlchen-test-dir/testuser/rotkehlchen_db_2020_05_02_12_33_05.backup', '/home/travis/.cache/.rotkehlchen-test-dir/testuser/dbinfo.json']

rotkehlchen/tests/utils/premium.py:207: AssertionError
```

## Task

Figure out why this happened in Travis and fix if indeed reproducible flaky test

--- a/rotkehlchen/db/dbhandler.py
+++ b/rotkehlchen/db/dbhandler.py
@@ -192,7 +192,7 @@ class DBHandler:
         self.disconnect()
         try:
             dbinfo = {'sqlcipher_version': self.sqlcipher_version, 'md5_hash': self.get_md5hash()}
-        except SystemPermissionError as e:
+        except (SystemPermissionError, FileNotFoundError) as e:
             # If there is problems opening the DB at destruction just log and exit
             log.error(f'At DB teardown could not open the DB: {str(e)}')
             return

--- a/rotkehlchen/tests/utils/premium.py
+++ b/rotkehlchen/tests/utils/premium.py
@@ -203,8 +203,16 @@ def assert_db_got_replaced(rotkehlchen_instance: Rotkehlchen, username: str):
     # Also check a copy of our old DB is kept around.
     directory = os.path.join(rotkehlchen_instance.data.data_directory, username)
     files = [os.path.join(directory, f) for f in os.listdir(directory)]
-    msg = f'Expected 2 files in the directory but got {files}'
-    assert len(files) == 2, msg
+    msg = f'Expected 2 or 3 files in the directory but got {files}'
+    assert len(files) == 2 or len(files) == 3, msg  # 3rd file is the dbinfo.json
     # The order of the files is not guaranteed
-    assert 'rotkehlchen.db' in files[0] or 'rotkehlchen.db' in files[1]
-    assert 'backup' in files[0] or 'backup' in files[1]
+    main_db_exists = False
+    backup_db_exists = False
+    for file in files:
+        if 'rotkehlchen.db' in file:
+            main_db_exists = True
+        elif 'backup' in file:
+            backup_db_exists = True
+
+    assert main_db_exists
+    assert backup_db_exists

